core/array/flatten_spec.rb

```diff
@@ -69,6 +69,12 @@
     [1, z, 6].flatten.should == [1, 2, 3, 4, 5, 6]
   end
 
+  it "does not call #to_ary on elements beyond the given level" do
+    obj = mock("1")
+    obj.should_not_receive(:to_ary)
+    [[obj]].flatten(1)
+  end
```

Comment:
Could add a version guard since this is only fixed from 2.3 and later? See this [example](https://github.com/sgarciac/spec/blob/33ed1b8445c1cf17792937829e4346945a726a27/core/enumerable/chunk_spec.rb#L86-L93).